### PR TITLE
Update the Kusto SDK Version from `2023-05-02` to `2023-08-15`

### PR DESCRIPTION
**Pull request by liuwuliuyun**
Details:
- Update the Kusto SDK Version from `2023-05-02` to `2023-08-15`
- Remove `ForceNew` from `virtual_network_configuration` block
- Add a `state` variable to `virtual_network_configuration` block, default `Enabled`. [Discussion on this newly added property](https://github.com/hashicorp/terraform-provider-azurerm/pull/22747)
- Remove `ForceNew` from `zones` property

TC Tests:

On Kusto:
image
The failed tests are also failed in main branch. Seperate PR like https://github.com/hashicorp/terraform-provider-azurerm/pull/22969 is raised to resolve faild accTests.

On DataShare Resource Provider
image

On Other Resource Provider
```
GOROOT=C:\Program Files\Go #gosetup
GOPATH=C:\Users\yunliu1\go #gosetup
"C:\Program Files\Go\bin\go.exe" test -c -o C:\Users\yunliu1\AppData\Local\JetBrains\GoLand2023.2\tmp\GoLand\___TestAccTimeSeriesDatabaseConnection_basic_in_github_com_hashicorp_terraform_provider_azurerm_internal_services_digitaltwins.test.exe github.com/hashicorp/terraform-provider-azurerm/internal/services/digitaltwins #gosetup
"C:\Program Files\Go\bin\go.exe" tool test2json -t C:\Users\yunliu1\AppData\Local\JetBrains\GoLand2023.2\tmp\GoLand\___TestAccTimeSeriesDatabaseConnection_basic_in_github_com_hashicorp_terraform_provider_azurerm_internal_services_digitaltwins.test.exe -test.v -test.paniconexit0 -test.run ^\QTestAccTimeSeriesDatabaseConnection_basic\E$ #gosetup
=== RUN   TestAccTimeSeriesDatabaseConnection_basic
=== PAUSE TestAccTimeSeriesDatabaseConnection_basic
=== CONT  TestAccTimeSeriesDatabaseConnection_basic
--- PASS: TestAccTimeSeriesDatabaseConnection_basic (1275.45s)
PASS


Process finished with the exit code 0

GOROOT=C:\Program Files\Go #gosetup
GOPATH=C:\Users\yunliu1\go #gosetup
"C:\Program Files\Go\bin\go.exe" test -c -o C:\Users\yunliu1\AppData\Local\JetBrains\GoLand2023.2\tmp\GoLand\___TestAccTimeSeriesDatabaseConnection_complete_in_github_com_hashicorp_terraform_provider_azurerm_internal_services_digitaltwins.test.exe github.com/hashicorp/terraform-provider-azurerm/internal/services/digitaltwins #gosetup
"C:\Program Files\Go\bin\go.exe" tool test2json -t C:\Users\yunliu1\AppData\Local\JetBrains\GoLand2023.2\tmp\GoLand\___TestAccTimeSeriesDatabaseConnection_complete_in_github_com_hashicorp_terraform_provider_azurerm_internal_services_digitaltwins.test.exe -test.v -test.paniconexit0 -test.run ^\QTestAccTimeSeriesDatabaseConnection_complete\E$ #gosetup
=== RUN   TestAccTimeSeriesDatabaseConnection_complete
=== PAUSE TestAccTimeSeriesDatabaseConnection_complete
=== CONT  TestAccTimeSeriesDatabaseConnection_complete
--- PASS: TestAccTimeSeriesDatabaseConnection_complete (1359.62s)
PASS


Process finished with the exit code 0
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -15,7 +15,7 @@ require (
 	github.com/google/go-cmp v0.5.9
 	github.com/google/uuid v1.3.1
 	github.com/hashicorp/go-azure-helpers v0.61.0
-	github.com/hashicorp/go-azure-sdk v0.20230918.1115907
+	github.com/hashicorp/go-azure-sdk v0.20230922.1111207
```
Comment (jackofallops) [inline go.mod:18]: Can we split this out into a separate PR please? We regenerate this regularly and need to ensure we don't create conflicts in the versions used.

**Review by jackofallops — CHANGES_REQUESTED** [review]
Hi @liuwuliuyun - Before we review, can you please split out the version bump of `go-azure-sdk`? We ask that this is done independently of other change due to the frequency of the releases if it.

Thanks

File: go.mod (hunk 1 above)
Comment (liuwuliuyun) [reply go.mod:18]: Ok, I will close this PR and reopen it after tests are running property.

**Comment by liuwuliuyun** [comment]
Close this till the go-azure-sdk version upgrade complete

**Comment by liuwuliuyun** [comment]
Reopen after sdk update

**Review by katbyte — CHANGES_REQUESTED** [review]
Hey @liuwuliuyun 

Given the schema changes you are making here could we please separate out the SDK upgrade and then schema changes you wish to make?

also, regarding:
`If the consumer is using terraform deployment with auto approve variable, he wouldn't see the "create new" warning and his cluster will be deleted.`

What is the reasoning here? That is the correct behaviour and by design - the property cannot be updated without replacing the resource therefore this change this would prevent users from updating the property and break other workflows.  users should be reviewing the planned changes prior to running `terraform apply`

**Comment by liuwuliuyun** [comment]
Hi @katbyte , thanks for the review comment. I will reopen a PR only for SDK update. Regarding the second part, we argue with the service team with our design principle and service team declare current period as `risk-period` just incase any user is not checking the plan message carefully and results in data loss. Closing this PR for now.
